libr/core/canal.c

```diff
@@ -3645,9 +3645,7 @@ R_API void r_core_anal_esil(RCore *core, const char *str, const char *target) {
 	//this is necessary for the hook to read the id of analop
 	ESIL->user = &op;
 	ESIL->cb.hook_mem_read = &esilbreak_mem_read;
-	if (!core->io->cached) {
-		ESIL->cb.hook_mem_write = &esilbreak_mem_write;
-	}
+	ESIL->cb.hook_mem_write = &esilbreak_mem_write;
```

Comment:
+1 some ppl will dislike it, but this is how it should be

Reply:
As a notice `esilbreak_mem_write` is not implemented.

I hate unnecessary checks, generally.